packages/components/src/icon/test/index.js

```diff
@@ -1,71 +1,67 @@
 /**
  * External dependencies
  */
-import { shallow } from 'enzyme';
-
-/**
- * WordPress dependencies
- */
-import { Component } from '@wordpress/element';
+import { render, screen } from '@testing-library/react';
 
 /**
  * Internal dependencies
  */
-import Dashicon from '../../dashicon';
 import Icon from '../';
 import { Path, SVG } from '../../';
 
 describe( 'Icon', () => {
+	const testId = 'icon';
 	const className = 'example-class';
 	const svg = (
 		<SVG>
 			<Path d="M5 4v3h5.5v12h3V7H19V4z" />
 		</SVG>
 	);
-	const style = { fill: 'red' };
 
 	it( 'renders nothing when icon omitted', () => {
-		const wrapper = shallow( <Icon /> );
+		render( <Icon data-testid={ testId } /> );
 
-		expect( wrapper.type() ).toBeNull();
+		expect( screen.queryByTestId( testId ) ).not.toBeInTheDocument();
 	} );
 
 	it( 'renders a dashicon by slug', () => {
-		const wrapper = shallow( <Icon icon="format-image" /> );
+		render( <Icon data-testid={ testId } icon="format-image" /> );
 
-		expect( wrapper.find( 'Dashicon' ).prop( 'icon' ) ).toBe(
-			'format-image'
+		expect( screen.getByTestId( testId ) ).toHaveClass(
+			'dashicons-format-image'
 		);
 	} );
 
 	it( 'renders a function', () => {
-		const wrapper = shallow( <Icon icon={ () => <span /> } /> );
+		render( <Icon icon={ () => <span data-testid={ testId } /> } /> );
 
-		expect( wrapper.name() ).toBe( 'span' );
+		expect( screen.getByTestId( testId ) ).toBeInTheDocument();
 	} );
 
 	it( 'renders an element', () => {
-		const wrapper = shallow( <Icon icon={ <span /> } /> );
+		render( <Icon icon={ <span data-testid={ testId } /> } /> );
 
-		expect( wrapper.name() ).toBe( 'span' );
+		expect( screen.getByTestId( testId ) ).toBeInTheDocument();
 	} );
 
 	it( 'renders an svg element', () => {
-		const wrapper = shallow( <Icon icon={ svg } /> );
+		render( <Icon data-testid={ testId } icon={ svg } /> );
 
-		expect( wrapper.name() ).toBe( 'SVG' );
+		expect( screen.getByTestId( testId ) ).toBeInTheDocument();
 	} );
 
 	it( 'renders an svg element with a default width and height of 24', () => {
-		const wrapper = shallow( <Icon icon={ svg } /> );
+		render( <Icon data-testid={ testId } icon={ svg } /> );
+		const icon = screen.queryByTestId( testId );
```

Comment:
We should be able to use `get*` instead of `query*` here, see [this for rationale](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence).

```suggestion
		const icon = screen.getByTestId( testId );
```